server/services/instrumenter.js

```diff
@@ -39,30 +41,49 @@ if (Conf.IS_COVERAGE_ACTIVE) {
         return instrumenter.instrument(content, path, callback);
     };
 
-    checkIfAutorised = function (filters, file) {
-        let allowed = true;
-        for (let i = 0; i < filters.length; i++) {
-            let filterRegex = new RegExp(filters[i], "i");
-            allowed = allowed && filterRegex.test(file) === false;
-            if (!allowed) {
-                return false;
-            }
-        }
-        return allowed;
-    };
+    shouldIgnore = function (filePath, isAServerSideFile) {
+
+      if(!Conf) return false;
+       if (Conf.include) {
+         if (Conf.include.some(pattern => fileMatch(pattern))) {
+           return false;
+         }
+       }
+
+       let shouldIgnore = false;
+       if (!Conf.exclude) return false;
+       if (Conf.exclude.general) {
+         shouldIgnore = Conf.exclude.general.some(pattern => fileMatch(pattern));
+       }
+
+       if (!shouldIgnore && Conf.exclude.server && isAServerSideFile) {
+         shouldIgnore = Conf.exclude.server.some(pattern => fileMatch(pattern));
+       }
+
+       if (!shouldIgnore && Conf.exclude.client && !isAServerSideFile) {
+         shouldIgnore = Conf.exclude.client.some(pattern => fileMatch(pattern));
+       }
+
+       return shouldIgnore;
+
+       function fileMatch(pattern) {
```

Comment:
It's not a problem as I see, this is not a function that should be accessible to the other functions, I think it should be very fine